### PR TITLE
DSD-1651: Documentation: sticky header accessibility guide

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes JIRA ticket [DSD-1651](https://jira.nypl.org/browse/DSD-1651)

## This PR does the following:
- Adds the Sticky Elements accessibility guide page

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (7emansell) [inline src/components/AccessibilityGuide/StickyElements.mdx]: Want to include a couple lines here about the design rationale for sticky elements

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (7emansell) [inline src/components/AccessibilityGuide/StickyElements.mdx]: As the comment says– don't know how much detail to give on this one implementation. It would be the most common one

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx]: Say "React hook" to be more precise.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx]: Use the `main` link https://github.com/NYPL/digital-collections/blob/main/src/hooks/useScrolled.ts

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx:74]: Can you add a type to `sticky` and include above what you need to pass to the function?

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [reply src/components/AccessibilityGuide/StickyElements.mdx]: This is great already! You can add a line to say something such as, "for the complete code and more information, check out..."

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [reply src/components/AccessibilityGuide/StickyElements.mdx]: Here are a few resources for sticky elements on sites:
- https://www.smashingmagazine.com/2023/05/sticky-menus-ux-guidelines/ 
- https://technology.blog.gov.uk/2018/05/21/sticky-elements-functionality-and-accessibility-testing/
- https://design.gs.com/patterns/sticky-elements

It looks like the general pattern is to use sticky headers/menus/elements when it is useful for users to understand where they are on a site, keep them within a certain process flow, or to convey information that we think will be useful.

But for NYPL, I'd argue that we should try to stay away from sticky components where possible and only use them when we need to convey critical or specific information. I think the DC sticky header is a feature that existed and we want parity from the old app to the new app for consistency and feel.

@bigfishdesign13 and I would need a bit more time for this "General Information" context.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx:40]: While waiting for the context in "General Information", there can be a new section titled "General Issues". You can start off by saying that if you absolutely must use sticky elements on a page, that you need to think about UX issues such as taking up space, breaking when zooming in on a mobile device, difficult to access when tabbing forward and backwards, and hiding focused elements (examples taken from here https://adamsilver.io/blog/the-problem-with-sticky-menus-and-what-to-do-instead/)

Then you can state the problem faced in the new DC homepage and then keep everything after "Sighted keyboard users....". It can be clear that you're focusing only on the sticky header hiding focused elements issue and not covering the other issues.

**Review by EdwinGuzman — COMMENTED** [review]
Added some comments. I think this may take a bit more back-and-forth but this is a great start. I'm coming from the approach where we say that we don't recommend it but understand there are cases where it's useful. If that's the case, look out for these UX issues. And then point out the specific issue you worked on and how you converted code into React to make the DC Header accessible.

**Comment by 7emansell** [comment]
Has approval from Clare O'Keeffe!

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx]: Say "new Digital Collections site" since you reference the old site at the end of the sentence.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [inline src/components/AccessibilityGuide/StickyElements.mdx]: Can you indent the code snippet here and below? It doesn't render well in storybook and makes it hard to read.

**Review by EdwinGuzman — COMMENTED** [review]
At the moment, I don't think we need to have an NYPL-specific policy. The intro and context is great already and we can add more later. The only thing to fix is the two code snippets indentation.

**Review by EdwinGuzman — APPROVED** [review]
Thanks! Let's wait on a review from Marty before merging.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/StickyElements.mdx:7]: When saying `Sticky Elements`, does that specifrically refer to elements where `position: sticky` has been applied?  Or does that also include `position: sticky`?

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/StickyElements.mdx]: I would like take some of this information and move it into a new section called `NYPL Patterns`.  Essentially, it's the second sentence that should be moved.  That sounds like an NYPL recommendation more than an overarching suggestion.  To keep this ticket moving forward, let's create another ticket to restructure this section.

Adittionally, let's try to write the general docs without using a first person or second person point of view.  The docs in the `NYPL Patterns` section might include some of that, but the rest should be very generic.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/StickyElements.mdx:52]: @EdwinGuzman Should we consider including the hook in the DS?

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/StickyElements.mdx]: I see now that you are making a distinction between `position: sticky` and `position: fixed`. Can you include a brief explanation telling why using `fixed` would be better for this situation?

**Review by bigfishdesign13 — COMMENTED** [review]
Looks good.  I left a few coments.  I think some of the comments should be addressed before this is merged, but others can be pushed into a new ticket.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (7emansell) [reply src/components/AccessibilityGuide/StickyElements.mdx:7]: I think I'm missing the distinction between these two ?

**Comment by 7emansell** [comment]
Thanks @bigfishdesign13, updated to remove 1st/2nd person voice. Let me know if what I left in is appropriate for this guide, or if more needs to be pulled out to the NYPL Patterns doc.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (bigfishdesign13) [reply src/components/AccessibilityGuide/StickyElements.mdx:7]: Oops.  That's a typo on my part.  I meant to say `position: fixed` in that last sentence.

**Review by bigfishdesign13 — APPROVED** [review]
This looks good.

File: src/components/AccessibilityGuide/StickyElements.mdx
Comment (EdwinGuzman) [reply src/components/AccessibilityGuide/StickyElements.mdx:52]: Not sure. I think it's fine as an app-only hook for now and if another team needs to create a sticky element on their site, we then move it over.

**Pull request merged**
